docs/lang/articles/advanced/layout.md

```diff
@@ -318,53 +301,34 @@ x = ti.field(dtype=ti.f32)
 fb1.dense(ti.ij, (5, 5)).place(x)
 fb1_snode_tree = fb1.finalize()  # Finalizes the FieldsBuilder and returns a SNodeTree
 func(x)
+...
+fb1_snode_tree.destroy() # Destruction
 
 fb2 = ti.FieldsBuilder()
 y = ti.field(dtype=ti.f32)
 fb2.dense(ti.i, 5).place(y)
 fb2_snode_tree = fb2.finalize()  # Finalizes the FieldsBuilder and returns a SNodeTree
 func(y)
+...
+fb2_snode_tree.destroy() # Destruction
 ```
+Actually, the above demonstrated `ti.root` statements are implemented with `FieldsBuilder`, despite that `ti.root` has the capability to automatically manage memory allocations and recycling.
 
-In fact, `ti.root` is implemented by `FieldsBuilder` implicitly, so you can
-allocate the fields directly under `ti.root`:
-```python
-import taichi as ti
-ti.init()  # Implicitly: ti.root = ti.FieldsBuilder()
+### Packed mode
 
-@ti.kernel
-def func(v: ti.template()):
-    for I in ti.grouped(v):
-        v[I] += 1
+By default, Taichi implicitly fits a field in a larger buffer with power-of-two dimensions. We take the power-of-two padding convention as it is widely adopted in computer graphics. The design enables fast indexing with bitwise arithmetics and better memory address alignment, while trades off memory occupations especially for large irregular fields.
```

Comment:
```suggestion
By default, Taichi implicitly fits a field in a larger buffer with power-of-two dimensions. We take the power-of-two padding convention because it is widely adopted in computer graphics. The design enables fast indexing with bitwise arithmetic and better memory address alignment, while trading off memory occupations especially for large irregular fields.
```